### PR TITLE
rpc: enforce rpc api module availability

### LINKED ISSUE #20467: --rpcapi does not cause or report an error when given an invalid parameter
#### System information

```
$ ./build/bin/geth version && uname -a
Geth
Version: 1.9.10-unstable
Git Commit: 6ae9dc15cc51310f33a011dcafab5739a6ceea93
Git Commit Date: 20191218
Architecture: amd64
Protocol Versions: [64 63]
Go Version: go1.12.1
Operating System: linux
GOPATH=/home/ia/go
GOROOT=/home/ia/go1.12.1.linux-amd64/go
Linux ubp52 5.2.9-050209-generic #201908160940 SMP Fri Aug 16 09:42:58 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
```

#### Expected behaviour

Attempting to configure `--rpcapi` with an invalid module should not be allowed.

```
> ./build/bin/geth --rpcapi eth,doesnotexist
#> RPC module 'doesnotexist' is not an available RPC module.
```

#### Actual behaviour

Geth doesn't complain when you pass an invalid module.

```
> ./build/bin/geth --rpcapi eth,doesnotexist
#> No complaints, geth just runs and ignores the invalid parameter.
```

### PR DESCRIPTION
**Pull request by meowsbits**
Fixes #20467

### CODE REVIEW

File: rpc/endpoints.go [hunk 1]
```diff
@@ -22,11 +22,37 @@ import (
...
+// available API services. If an invalid module is given (ie API "foo" wanted which does not exist),
+// then log.Crit is used to cause program to exit, logging the invalid module and a list of available
+// API service names.
+func mustAvailableModule(module string, apis []API) {
+	for _, api := range apis {
+		if module == api.Namespace {
+			return
+		}
+	}
+	log.Crit("invalid api module", "module", module, "available", func() (available []string) {
```
Comment (fjl) [inline rpc/endpoints.go]: Please change the code to return an error instead of calling `log.Crit`.

Comment (meowsbits) [reply rpc/endpoints.go]: 3ac3773

**Comment by fjl** [comment]
Sorry for taking so long to get to this. I have improved this change a little bit in #20597. The error message is now shorter and checking availability no longer has quadratic complexity.

**Comment by fjl** [comment]
Thank you for the initiative on this issue!

**Pull request closed without merging**